bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/items/ItemUIRegistry.java

```diff
@@ -51,10 +68,10 @@ public interface ItemUIRegistry extends ItemRegistry, ItemUIProvider {
      * current value of the item and padded by a "<span>" element.
      *
      * @param w the widget to retrieve the label for
-     * @return the label to use for the widget
+     * @return the label to use for the widget, together with its source
      */
     @Nullable
-    String getLabel(Widget w);
+    WidgetLabelWithSource getLabel(Widget w);
```

Comment:
Maybe you can avoid introducing this serious breaking change ?
If not, Basic UI will certainly no more compile.

Reply:
Well, I *can* avoid it by adding a second method `getLabelSource`, but the basic question is do we prefer backwards compatibility or API design here? If the latter, I can also provide a patch to adjust BasicUI to the changed API.

Reply:
I have splitted retrieving the label source to a separate method as mentioned. This means the 'retrieve label' logic now needs to run twice for each sitemap widget or SSE event (once for actual label, a second time for source), but this shouldn't be too much of an issue.